### ISSUE TITLE
Coupon model not loaded (fatal error)

### ISSUE THREAD

**Issue by gelderblominternet** [mollie/OpenCart#248]
There is an error inside catalog/controller/mollie.php if you checkout with a coupon. This error effects OC 2.3 and 3.x.

```
if(version_compare(VERSION, '2.0.3.1', '<=')) {
    $coupon = $this->load->model('checkout/coupon');
} else {
    $coupon = $this->load->model('total/coupon');
}
```

Should be replaced with:

```
if (version_compare(VERSION, '2.2', '<')) {
    $coupon = $this->load->model('total/coupon');
} elseif (version_compare(VERSION, '2.3', '<')) {
    $coupon = $this->load->model('total/coupon');
} else {
    $coupon = $this->load->model('extension/total/coupon');
}
```

I can create a pull request. Please let me know.

**Comment by QualityWorks** [comment]
We are working on a new release that will have fixed this

**Comment by gelderblominternet** [comment]
Thanks for the response!

**Comment by gelderblominternet** [comment]
My example runs in an error after submitting the order. This is the fix I have now:

```
if (version_compare(VERSION, '2.2', '<')) {
    $this->load->model('checkout/coupon');
    $coupon_info = $this->model_checkout_coupon->getCoupon($this->session->data['coupon']);
} elseif (version_compare(VERSION, '2.3', '<')) {
    $this->load->model('total/coupon');
    $coupon_info = $this->model_total_coupon->getCoupon($this->session->data['coupon']);
} else {
    $this->load->model('extension/total/coupon');
    $coupon_info = $this->model_extension_total_coupon->getCoupon($this->session->data['coupon']);
}

// $coupon->getCoupon($this->session->data['coupon']);
```

**Comment by QualityWorks** [comment]
we are preparing a release for later today that will fix this

**Comment by QualityWorks** [comment]
this should all be fixed in 10.1.0, just upload the files from the zip, and optionally, remove the files from storage/modification if you are experiencing errors. (and after you login again, refresh modifications)
